### PR TITLE
Adjust method channel mechanism on iOS

### PR DESCRIPTION
**Pull request by Swag-Tristan**
To align the mechanism with Android side.

- [x] Remove invoking flutter channel from Swift, use `FlutterResult` instead
- [x] Add missing flutter result call for the initialization check

### CODE REVIEW

File: ios/Classes/SwiftZendeskMessagingPlugin.swift [hunk 1]
```diff
@@ -23,92 +23,114 @@ public class SwiftZendeskMessagingPlugin: NSObject, FlutterPlugin {
             self.processMethodCall(call, result: result)
         }
     }
-
+    
     private func processMethodCall(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
         let method = call.method
         let arguments = call.arguments as? Dictionary<String, Any>
-        let zendeskMessaging = ZendeskMessaging(flutterPlugin: self, channel: channel)
+        let zendeskMessaging = ZendeskMessaging(flutterPlugin: self)
```
Comment (JimmyTai) [inline ios/Classes/SwiftZendeskMessagingPlugin.swift:30]: I think u don't want to create ZendeskMessaging every time received a method channel call.

File: ios/Classes/SwiftZendeskMessagingPlugin.swift [hunk 2]
```diff
@@ -23,92 +23,114 @@ public class SwiftZendeskMessagingPlugin: NSObject, FlutterPlugin {
...
+            }
+            zendeskMessaging.show(rootViewController: UIApplication.shared.delegate?.window??.rootViewController, flutterResult: result)
+        case "loginUser":
+            if (!isInitialized) {
+                print("\(TAG) - Messaging needs to be initialized first.\n")
+                reportNotInitializedFlutterError(result: result)
+                return
+            }
+            let jwt: String = arguments?["jwt"] as! String
+            zendeskMessaging.loginUser(jwt: jwt, flutterResult: result)
```
Comment (JimmyTai) [inline ios/Classes/SwiftZendeskMessagingPlugin.swift:50]: This casting is not safe.

File: ios/Classes/SwiftZendeskMessagingPlugin.swift [hunk 3]
```diff
@@ -23,92 +23,114 @@ public class SwiftZendeskMessagingPlugin: NSObject, FlutterPlugin {
...
+            result(FlutterMethodNotImplemented)
         }
     }
-
-    private func handleMessageCount() ->Int{
-         let zendeskMessaging = ZendeskMessaging(flutterPlugin: self, channel: channel)
-
+    
+    private func handleMessageCount() -> Int {
+        let zendeskMessaging = ZendeskMessaging(flutterPlugin: self)
```
Comment (JimmyTai) [inline ios/Classes/SwiftZendeskMessagingPlugin.swift:117]: Please reuse `ZendeskMessaging`.

**Review by JimmyTai — APPROVED** [review]

File: ios/Classes/SwiftZendeskMessagingPlugin.swift (hunk 1 above)
Comment (Swag-Tristan) [reply ios/Classes/SwiftZendeskMessagingPlugin.swift]: Done

File: ios/Classes/SwiftZendeskMessagingPlugin.swift (hunk 3 above)
Comment (Swag-Tristan) [reply ios/Classes/SwiftZendeskMessagingPlugin.swift]: Done

File: ios/Classes/SwiftZendeskMessagingPlugin.swift (hunk 2 above)
Comment (Swag-Tristan) [reply ios/Classes/SwiftZendeskMessagingPlugin.swift]: Use the soft unwrap instead.

**Pull request merged**